src/parser/deserializer.ts

```diff
@@ -314,23 +314,16 @@ function deserializeObject(
         (buffer[index + 1] << 8) |
         (buffer[index + 2] << 16) |
         (buffer[index + 3] << 24);
-      let arrayOptions = options;
+      let arrayOptions: DeserializeOptions = options;
 
       // Stop index
       const stopIndex = index + objectSize;
 
       // All elements of array to be returned as raw bson
       if (fieldsAsRaw && fieldsAsRaw[name]) {
-        arrayOptions = {};
-        for (const n in options) {
-          (
-            arrayOptions as {
-              [key: string]: DeserializeOptions[keyof DeserializeOptions];
-            }
-          )[n] = options[n as keyof DeserializeOptions];
-        }
-        arrayOptions['raw'] = true;
+        arrayOptions = { ...options, raw: true };
```

Comment:
The issue is related because it has similar surprising side effects when there are keys that you didn't specify yourself effecting the outcomes of your deserialize calls. I tried demonstrating the possible buggy behavior in the test in `test/node/parser/deserializer.test.ts`. If the global object prototype has been polluted with unexpected keys they can control the behavior of the BSON library without actually needing access to the options object a user passes in.

Reply:
I see.  This impacts all other options too, no? 

```typescript
import { deserialize, serialize } from "./src/bson";

const bytes = serialize({ someKey: { 
    nested: "value"
} });

const opts = {} 
Object.setPrototypeOf(opts, { raw: true });

const result = deserialize(bytes, opts);

console.log(result); 
// { someKey: <Buffer 17 00 00 00 02 6e 65 73 74 65 64 00 06 00 00 00 76 61 6c 75 65 00 00> }
```

It seems like what we'd want to do is only look for own properties on the options objects too when accessing individual fields (not in scope for this work).  My 2-cents would be to remove this from this PR and file a follow up to only consider options that are set as own properties (probably an easy fix by making a shallow clone of the options before we begin any work deserializing / serializing).

If you want to leave this in this PR, that's also okay because technically the scope of the ticket is "Use Object.keys to enumerate keys on a JS object passed to BSON", and this is an instance where we're using a for-in loop.  If we leave this work in this PR, we should still file a follow up ticket I think.  And I'll comment below, but I think we can make the test easier to understand.

Reply:
Works for me.  I left a comment about the test below